packages/lodestar/src/chain/regen/regen.ts

```diff
@@ -152,7 +180,7 @@ export class StateRegenerator implements IStateRegenerator {
     const blocksToReplay = [block];
     let state: CachedBeaconState<allForks.BeaconState> | null = null;
     for (const b of this.forkChoice.iterateBlockSummaries(block.parentRoot)) {
-      state = this.stateCache.get(b.stateRoot);
+      state = this.stateCache.get(b.stateRoot, rmetrics);
```

Comment:
Instead of passing a full metrics instance as a function argument, why not pass a `IRegenCaller`? You can pass the metrics object through the state cache constructor

Reply:
actually, i abstracted out the cache metrics to be agnostic of the caller and pass it the object in which labels are pre bundeled, even when it could be called outside the regen modules otherwise it gets deeply tied to rcaller.

Reply:
I would remove all labels from the state cache metrics, see https://github.com/ChainSafe/lodestar/pull/2852#discussion_r686155282